### DISCUSSION TITLE
Question: how to render a template from code?

### DISCUSSION THREAD

**Discussion by szilardcsere89** [OrchardCMS/OrchardCore#7349] [Q&A]
I have created a template:
![image](https://user-images.githubusercontent.com/5823033/96859106-65f6fa00-1461-11eb-8569-c3b963ed45ab.png)

I'm trying the following but I get errors:
```
var shape = await _shapeFactory.CreateAsync("SomeShape", new { MyData = 100 });
await displayHelper.ShapeExecuteAsync(shape);
```

What am I doing wrong?

**Comment by deanmarcussen** [comment]
I imagine you are getting a Shape Not Found exception, but it's hard to say because you don't say what errors you are getting ;)

Probably you need to make a shape template in code, i.e. `SomeShape.liquid` or `SomeShape.cshtml` before you can override it with a template in the ui.

**Comment by szilardcsere89** [comment]
Yes: `Exception: Shape type 'SomeShape' not found` is thrown. The shape `SomeShape.cshtml` is not overridden even if the template is created.

**Comment by deanmarcussen** [comment]
Where are you placing your `SomeShape` file? 

A `Views` folder in a theme? or somewhere else?

**Reply by szilardcsere89** [reply]
`View` folder in the module. I already know what is the problem.
